sympy/matrices/matrices.py

```diff
@@ -783,6 +784,23 @@ def __repr__(self):
         return "MatrixKind(%s)" % self.element_kind
 
 
+@Mul._kind_dispatcher.register(MatrixKind, type(NumberKind))
+def _(k1, k2):
+    """Return MatrixKind. Its element kind is selected by recursive dispatching."""
+    # Deal with Mul._kind_dispatcher's commutativity
+    if isinstance(k1, MatrixKind):
+        mk = k1
+    else:
+        mk = k2
```

Comment:
Why do we need an `isinstance` check if we are using multiple dispatch? The whole point of multiple dispatch is that it selects the method to call based on `isinstance`...